src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+
+/// \cond
+class DataVector;
+namespace gsl {
+template <class T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace GeneralizedHarmonic::BoundaryConditions {
+namespace detail {
+// @{
+/*!
+ * \brief Computes the expression needed to set bnudary conditions on Dt<VPsi>
+ *
+ * \details Computes the correction to the characteristic projected
+ * time-derivatives of the fundamental variables corresponding to \f$u^\psi\f$:
+ *
+ * \f{align}
+ * \partial_t u^{\psi}_{ab} = \partial_t u^{\Psi}_{ab} + v_{\psi} n^i C_{iab}
+ * \f}
+ *
+ * where \f$n^i\f$ is the local unit normal to the external boundary,
+ * and \f$C_{iab} = \partial_i \psi_{ab} - \Phi_{iab}\f$ is the
+ * three-index constraint.
+ */
+template <size_t VolumeDim, typename DataType>
+void set_dt_v_psi_constraint_preserving(
```

Comment:
something to consider, not a change request: the way that the Bjorhus boundary conditions are implement is you specify a _correction_ that gets added to the `dt evolved_vars` in the volume. This means it might be easier to have this function only compute `+ v_{\psi} n^i C_{iab}`. You can, of course, change from `dt psi` -> `dt psi^{BC}` by just subtracting off the volume time derivative, so it's not a big issue, just an FYI :)

Reply:
Currently this and associated PRs were aiming or just subtracting off the volume time derivative, but I think just calculating the correction will save computation in the longer run. I've modified the function here to just calculate the correction term, and will do so in other PRs as well.